src/rules/react-usememo-should-be-component.ts

```
@@ -29,6 +29,539 @@ const isJsxElement = (node: TSESTree.Node | null): boolean => {
+  };
+
+  /**
+   * Find a function definition by name in the current scope
+   */
+  const findFunctionDefinition = (
+    functionName: string,
+    contextNode: TSESTree.Node,
+  ): TSESTree.Node | null => {
+    // Walk up to find the containing function/block scope
+    let current: TSESTree.Node | undefined = contextNode;
+    while (current && current.parent) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)